cpp/log/file_db-inl.h

```diff
@@ -62,36 +73,27 @@ typename Database<Logged>::WriteResult FileDB<Logged>::CreateSequencedEntry_(
   cert_trans::ScopedLatency latency(
       latency_by_op_ms.GetScopedLatency("create_sequenced_entry"));
 
-  std::unique_lock<std::mutex> lock(lock_);
-  util::StatusOr<std::string> old_hash(
-      HashFromIndex(lock, logged.sequence_number()));
-  const std::string new_hash(logged.Hash());
-
-  if (old_hash.ok()) {
-    // Same log entry resubmitted, this is fine.
-    if (new_hash == old_hash.ValueOrDie()) {
-      return this->OK;
-    }
-
-    LOG(WARNING) << "Attempting to re-use sequence number "
-                 << logged.sequence_number() << " for entry: " << new_hash
-                 << " (existing: (" << old_hash.ValueOrDie() << "):\n"
-                 << logged.DebugString();
-
-    return this->SEQUENCE_NUMBER_ALREADY_IN_USE;
-  }
-
   std::string data;
   CHECK(logged.SerializeToString(&data));
 
+  const std::string seq_str(FormatSequenceNumber(logged.sequence_number()));
+
+  std::unique_lock<std::mutex> lock(lock_);
+
   // Try to create.
-  util::Status status(cert_storage_->CreateEntry(new_hash, data));
+  util::Status status(cert_storage_->CreateEntry(seq_str, data));
   if (status.CanonicalCode() == util::error::ALREADY_EXISTS) {
-    return this->ENTRY_ALREADY_LOGGED;
+    std::string existing_data;
+    status = cert_storage_->LookupEntry(seq_str, &existing_data);
+    CHECK_EQ(status, util::Status::OK);
```

Comment:
Did I put in a `CHECK_OK`? This is fine too (the important is that it prints the status if it's not OK).